tests/testing_harness.py

```diff
@@ -383,6 +383,58 @@ def _get_results(self):
         """Digest info in the statepoint and return as a string."""
         return super()._get_results(True)
 
+class TolerantPyAPITestHarness(PyAPITestHarness):
```

Comment:
Can you add some documentation to this class briefly explaining its purpose and an example of when it should be used for testing?

Reply:
Good thinking -- I've added in a comment there to explain when it may be useful. Basically as the random ray solver uses FP32, you do see some changes in the last digit due to rounding (when running in parallel) due to floating point non-associativity. If the order that rays contribute to the flux each iteration changes due to random thread scheduling, then the roundoff also changes slightly.

Reply:
Hm, we should use this elsewhere! I always see CMFD tests fail due to differing linear algebra backend versions. But that can be for the future.